book_extractors/tests/test_extraction_pipeline.py

```diff
@@ -0,0 +1,125 @@
+import pytest
+from book_extractors.common.extractors.base_extractor import BaseExtractor
+from book_extractors.extraction_pipeline import ExtractionPipeline, configure_extractor
+
+'''
+TODO: Populate this file with many more tests for extraction_pipeline
+'''
+
+
+class TestExtractionPipelineDataPassing:
+    def should_contain_parent_data_deep_in_the_subpipelines(self):
+        test_pipeline = ExtractionPipeline([
+            configure_extractor(MockExtractor),
+            configure_extractor(MockExtractorAWithSubpipeline)
+        ])
+        parent_data = {
+            'extraction_results': {
+                'result': 'test'
+            },
+            'metadata': {
+                'identity': 'i hate meta'
+            },
+            'parent_data': {
+                'extraction_results': {
+                    'name': {
+                        'gender': 'Male'
+                    },
+                    'test_result': 'testiness',
+                    'farmDetails': {
+                        'farmTotalArea': 57.0
+                    }
+                },
+                'metadata': {
+                    'identity': 'i love meta'
+                },
+                'parent_data': None
+            }
+        }
+
+        test_pipeline.process({'text': 'test string'}, parent_pipeline_data=parent_data)
```

Comment:
Jees. Tää on ok näin nyt. Sitten kun nää kaikki alkaa olla kohdallaan, niin korvataan tää koko pipelinea yksikkönä testaavana testinä. Voidaan jutella siitä sitten tarkemmin jos mun aiemmista sepustuksista ei auennut ja/tai jos jokin muuttuu oleellisesti noissa myöhemmissä vaiheissa 👍